src/state/opamFileTools.ml

```diff
@@ -1328,6 +1328,35 @@ let read_opam dir =
       (OpamPp.string_of_bad_format (OpamPp.Bad_format (snd err)));
     None
   | None, None -> None
+  | exception OpamPp.Bad_version ((_, _errmsg), Some version) ->
+    let sversion = OpamVersion.to_string version in
+    log "opam-version %S unsupported on %s. Added as dummy unavailable package."
+      sversion (OpamFile.to_string opam_file);
+    if OpamVersion.compare OpamVersion.current version >= 0 then begin
+      log
+        "Unexpected failure: failed to parse version %S \
+         but is below the current version %S"
+        sversion (OpamVersion.to_string OpamVersion.current);
+      Some
+        (OpamFile.OPAM.empty
+         |> OpamFile.OPAM.with_available (FBool false)
+         |> OpamFile.OPAM.with_descr_body
+           (Printf.sprintf
+              "Failed to parse version %S but is below the current version %S"
+              sversion (OpamVersion.to_string OpamVersion.current)))
```

Comment:
how can we reach that branch ?

Reply:
i don't know but i think it's worth the special case if it ever happens. As the log shows it is an "unexpected failure". Although if you prefer we can replace this by an assert

Reply:
I'm fine with the current code, it's just that i tried to raise that unexpected failure and i wasn't able to :)

Reply:
I've removed that section. Actually we do similar things in both cases and the description of the package added in the normal case does encompass this first one so it's not needed

Reply:
In fact, we should keep the version check, and if it is an unexpected error fail with internal error, we shouldn't act as it is known normal error.
From what i saw, there is no path via opam binary that lead to that path, but via lib, we can't be sure.

Reply:
I don't think it needs it. In both cases it's an unsupported version so i think we should have the same behaviour in both cases.